src/core/idhash.c

```diff
@@ -61,6 +61,18 @@ nni_id_map_fini(nni_id_map *m)
 	}
 }
 
+void
+nni_id_map_foreach(nni_id_map *m, nni_cb cb)
```

Comment:
I haven't looked at your usage yet, but I suspect it would be useful to have a different callback signature, one that provided both the id, and the value stored.

Reply:
done